### DISCUSSION TITLE
Handle abort message in ImageMagick Legacy

### DISCUSSION THREAD

**Discussion by cedeymard9** [ImageMagick/ImageMagick#6354] [Help]
### ImageMagick version

6.9.10-23

### Operating system, version and so on

Linux

### Description

Hi all, 

I try to understand a different behavior between 2 version of ImageMagick when I try to convert (resize) a jpg file : 
- on my Linux version (6.9.10-23), I have a "Abort" message and nothing is done
- on my Mac version (6.9.9-26), I have a message "convert: color profile operates on another colorspace `icc' @ error/profile.c/ProfileImage/994." and the image is created.

For information, my command is 
`convert -profile EuroscaleCoated.icc image.jpg -profile sRGB_Profile.icc -geometry 200x200 test.jpg`

Is there a way that the 6.9.10-23 version (Linux) could write a warning message but still create the image in output ? Maybe a force option or something with the configuration ?

Thanks for your answer

**Comment by snibgo** [comment]
I suggest you put the options in the order you want them executed. Read the input, then assign or convert to EuroscaleCoated, then convert to sRGB_Profile, then resize, then write the output.

I assume your intention for "`-geometry 200x200`" is to resize the image so it fits into a box that is 200x200 pixels. I suggest you use "`-resize 200x200`" instead.

So the command becomes:
```
convert image.jpg -profile EuroscaleCoated.icc -profile sRGB_Profile.icc -resize 200x200 test.jpg
```
I suppose `EuroscaleCoated.icc` is a CMYK profile. Is your input `image.jpg` also CMYK? Does it have an embedded ICC profile?

**Reply by cedeymard9** [reply]
Thanks for your quick answer. My main problem is more on the different behavior between those 2 versions, but I can explain the problem of this conversion : 
I check my input to know about CMYK or RGB wih exiftool -s. And on my image (see attached), I have information that seems not compatible for me : 

- ColorMode : CMYK
- ColorSpace : sRGB
- ICCProfileName : ISO Coated v2 300% (ECI)

So with my algorithm, I assume that my image is CMYK (because I check ColorMode first and then ColorSpace tag). Then I launch my conversion with my CMYK profile (EuroscaleCoated.icc). If I delete this parameters or choose another RGB profile, the conversion is ok on both version (but maybe color render is not really right).

My detection is certainly not perfect, but match on many others files.

In the first version, the image is well created, on the other version, I didn't have an image created, so that's a problem for me.

![logo](https://github.com/ImageMagick/ImageMagick/assets/66676689/634b1420-6e40-4fd6-a066-ac3b73b40e96)

**Comment by snibgo** [comment]
The file is weird. I have named it `amdRyzen.jpg`. Exiftool is contradictory:
```
Color Space                     : sRGB
ICC Profile Name                : ISO Coated v2 300% (ECI)
Color Mode                      : CMYK
```
"`ISO Coated`" is a CMYK profile, which contradicts the sRGB colorspace.

`-G1` explains the contradiction;
```
C:\web\im>exiftool -G1 amdRyzen.jpg

[ExifIFD]       Color Space                     : sRGB
[XMP-photoshop] ICC Profile Name                : ISO Coated v2 300% (ECI)
[XMP-photoshop] Color Mode                      : CMYK
```
The CMYK stuff is in a Photoshop XMP profile. IM will pass this profile to the output file, unless `-strip` is used, but otherwise ignores it.

So IM doesn't see any ICC profile or CMYK.

So IM thinks the image is sRGB, with no embedded profile. So your first "`-profile EuroscaleCoated.icc`" assigns a CMYK profile to an image that uses a RGB color model. This is bad news, partly because CMYK has 4 channels but RGB has only 3. Trying to convert from that profile to `sRGB_Profile.icc won't do anyhing helpful.

For this file, I suggest stripping the XMP profile so we don't have a contradiction, simply a sRGB image:
```
convert image.jpg -resize 200x200 test.jpg
```

**Comment by cedeymard9** [comment]
Thanks for your help @snibgo , for this specific image, the solution was to strip the XMP metadata. 

But I first look for a way to generate image even if I had this error. I still don't know if the different behavior is caused by the version (6.9.10-23 vs 6.9.9-26) or the platform (Linux vs Mac). I think there isn't a quick solution on that
